File: experiments/esbmc_python_c_poc/simple.c
#include <assert.h>

int nondet_int(void);

int f(int x)
{
    if (x < 100)
        return x + 1;

    return 0;
}

int main(void)
{
    int x = nondet_int();

    __ESBMC_assume(x >= -1000);
    __ESBMC_assume(x <= 1000);

    int r = f(x);

    assert(
        (x < 100 && r == x + 1) ||
        (x >= 100 && r == 0)
    );

    return 0;
}

File: experiments/esbmc_python_c_poc/simple.py
def f(x: int) -> int:
    if x < 100:
        return x + 1
    return 0


x: int = nondet_int()

__ESBMC_assume(x >= -1000)
__ESBMC_assume(x <= 1000)

r: int = f(x)

assert (x < 100 and r == x + 1) or \
       (x >= 100 and r == 0)
